packages/code-infra/src/utils/github.mjs

```diff
@@ -0,0 +1,321 @@
+/**
```

Comment:
A bunch of functions here have been created explicitly because some of the relevant octokit packages don't support our use case of storing access tokens.

There's `@octokit/auth-oath-device` but it asks to authenticate everytime a new `Octokit()` instance gets created.
There's also `@octokit/auth-app` but its options don't support a github app with just the `clientId`. It also requires `clientSecret` and a bunch of other options not really relevant and something we don't want to expose in code.